x/staking/keeper/delegation.go

```diff
@@ -755,17 +769,20 @@ func (k Keeper) BeginRedelegation(
 	return completionTime, nil
 }
 
-// CompleteRedelegation completes the unbonding of all mature entries in the
-// retrieved unbonding delegation object.
-func (k Keeper) CompleteRedelegation(
+// CompleteRedelegation completes the redelegations of all mature entries in the
+// retrieved redelegation object and returns the total redelegation (initial)
+// balance or an error upon failure.
+func (k Keeper) CompleteRedelegationWithAmount(
```

Comment:
Introducing a new function instead of replacing the old as we want to get this into the next 0.37 and 0.38 point releases.